src/vester.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import os
+import requests
+
+from hexbytes import HexBytes
+from web3 import Web3
+
+from config.constants import ARB_BADGER
+from config.constants import ARB_VESTER
+from config.constants import GAS_LIMITS
+from config.enums import Network
+from src.tx_utils import get_effective_gas_price
+from src.tx_utils import get_gas_price_of_tx
+from src.tx_utils import get_priority_fee
+from src.web3_utils import confirm_transaction
+from src.discord_utils import get_hash_from_failed_tx_error
+from src.discord_utils import send_success_to_discord
+from src.discord_utils import send_error_to_discord
+from src.utils import get_abi
+
+MAX_GAS_PRICE = int(1000e9)  # 1000 gwei
+CHAIN_CURRENCY = {Network.Arbitrum: ARB_BADGER}
+
+
+class Vester:
+    def __init__(
+        self,
+        chain: Network,
+        discord_url: str,
+        keeper_address=os.getenv("KEEPER_ADDRESS"),
+        keeper_key=os.getenv("KEEPER_KEY"),
+        base_oracle_address: str = os.getenv("ETH_USD_CHAINLINK"),
+        vesting_contract_address: str = ARB_VESTER,
+        node_url=os.getenv("ETH_NODE_URL"),
+    ):
+        self.logger = logging.getLogger(__name__)
```

Comment:
What's the point of assigning logger as instance attribute instead of defining it on the module level as it is done everywhere?